h/views/accounts.py

```diff
@@ -551,8 +555,10 @@ def post(self):
         )
 
     def _update_notifications(self, appstruct):
-        for notification in self._user_notifications():
-            notification.active = notification.type in appstruct["notifications"]
+        self.subscription_svc.set_subscriptions(
```

Comment:
Same here, if we iterate over `appstruct["notifications"]` it might be easier to get what the values are.